### DISCUSSION TITLE
abs/fabs instruction missing

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#297] [Q&A]
René pointed out that the ABS/FABS instructions are missing. It's no big deal to re-introduce them, but the question is how copes the compiler with ABS. Does it distinguish between floating-point and integer and call IABS and FABS or is there just one instruction and we have to distinguish internally for the types?

**Comment by rvjansen** [comment]
I was referring to the BIF - I put in an abs() that does not need the instructions. I can imagine that for assembler programs this comes in handy, but for Rexx not really needed.

**Reply by Peter-Jacob** [reply]
thanks, now you say it it's obvious - twisted mind.

**Reply by Peter-Jacob** [reply]
now I am confused! There is an ABS BIF in f0038, granted today I'd code it differently:

```
options levelb
abs: procedure = .string
arg number = .string
if left(number,1) = '-' then number = substr(number,2)
return number

substr: procedure = .string
arg string1 = .string, start = .int, len = length(string1) + 1 - start, pad = ' '
  
left: procedure = .string
arg string = .string, length1 = .int, pad = '0'

length: procedure = .int
  arg string1 = .string

```

**Comment by rvjansen** [comment]
That is the one I put there (because I needed it for something)
